File: Unidad3/C#/Programa21.cs
using System;

class Program
{
    static void Main()
    {
        Console.Write("Ingrese la cantidad de números naturales positivos: ");
        int cantidadNumeros = int.Parse(Console.ReadLine);

        for (int i = 1; i <= cantidadNumeros; i++)
        {
            Console.Write("Ingrese un número natural positivo: ");
            int numero = int.Parse(Console.ReadLine);

            if (numero > 0)
            {
                double cubo = Math.Pow(numero, 3);
                Console.WriteLine("El cubo de " + numero + " es: " + cubo);
            }
            else
            {
                Console.WriteLine("El número ingresado no es válido. Debe ser un número natural positivo.");
                i--; // Restar 1 al contador para volver a ingresar un número válido.
            }
        }
    }
}

File: Unidad3/Java/Programa21.java
package com.mycompany.programa21;

import java.util.Scanner;
public class Programa21 {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int cantidadNumeros;
        int suma = 0;

        System.out.print("Ingrese la cantidad de números a procesar: ");
        cantidadNumeros = input.nextInt();

        for (int i = 1; i <= cantidadNumeros; i++) {
            System.out.print("Ingrese un número natural positivo: ");
            int numero = input.nextInt();

            if (numero > 0) {
                suma += numero;
            } else {
                System.out.println("El número ingresado no es válido. Debe ser un número natural positivo.");
                i--; 
            }
        }

        System.out.println("La suma de los " + cantidadNumeros + " números naturales positivos ingresados es: " + suma);
    }
}